src/NServiceBus.Testing.Fakes/Fakes/TestableInvokeHandlerContext.cs

```diff
@@ -0,0 +1,83 @@
+﻿namespace NServiceBus.Testing
+{
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+    using NServiceBus.Persistence;
+    using NServiceBus.Pipeline;
+    using NServiceBus.Unicast.Messages;
+
+    /// <summary>
+    /// A testable implementation of <see cref="IInvokeHandlerContext" />.
+    /// </summary>
+    public class TestableInvokeHandlerContext : TestableIncomingContext, IInvokeHandlerContext
+    {
+        /// <summary>
+        /// Creates a new instance of <see cref="TestableInvokeHandlerContext" />.
+        /// </summary>
+        public TestableInvokeHandlerContext(IMessageCreator messageCreator = null) : base(messageCreator)
+        {
+        }
+
+        /// <summary>
+        /// Indicates if <see cref="IMessageHandlerContext.DoNotContinueDispatchingCurrentMessageToHandlers" /> has been called.
+        /// </summary>
+        public bool DoNotContinueDispatchingCurrentMessageToHandlersWasCalled { get; set; }
+
+        /// <summary>
+        /// Moves the message being handled to the back of the list of available
+        /// messages so it can be handled later.
+        /// </summary>
+        public virtual Task HandleCurrentMessageLater()
+        {
+            HandleCurrentMessageLaterWasCalled = true;
+            return Task.FromResult(0);
+        }
+
+        /// <summary>
+        /// Indicates if <see cref="IMessageHandlerContext.HandleCurrentMessageLater" /> has been called.
+        /// </summary>
+        public bool HandleCurrentMessageLaterWasCalled { get; set; }
```

Comment:
For testability reasons I'd leave those public by default. Gives users all
the flexibility. Don't apply usual review thinking here @andreasohlund

On Thursday, March 24, 2016, Andreas Öhlund <notifications@github.com>
wrote:

> In src/NServiceBus.Testing.Fakes/Fakes/TestableInvokeHandlerContext.cs
> <https://github.com/Particular/NServiceBus/pull/3629#discussion_r57307965>
> :
>
> > +        public bool DoNotContinueDispatchingCurrentMessageToHandlersWasCalled { get; set; }
> > +
> > +        /// <summary>
> > +        /// Moves the message being handled to the back of the list of available
> > +        /// messages so it can be handled later.
> > +        /// </summary>
> > +        public virtual Task HandleCurrentMessageLater()
> > +        {
> > +            HandleCurrentMessageLaterWasCalled = true;
> > +            return Task.FromResult(0);
> > +        }
> > +
> > +        /// <summary>
> > +        /// Indicates if <see cref="IMessageHandlerContext.HandleCurrentMessageLater" /> has been called.
> > +        /// </summary>
> > +        public bool HandleCurrentMessageLaterWasCalled { get; set; }
>
> Since we have HandleCurrentMessageLater can we make the setter private?
>
> —
> You are receiving this because you are on a team that was mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/Particular/NServiceBus/pull/3629/files/2239b5df4fbdf356a8e4c578abba81147cf5a8a1#r57307965>
>